inbc-program/tests/unit/test_source_app_parser.py

```diff
@@ -24,8 +26,76 @@ def test_parse_add_arguments_successfully(self):
                                      'deb-src http://example.com/ focal-security main'])
         self.assertEqual(f.filename, 'intel-gpu-jammy.list')
 
+    def test_parse_add_arguments_deb822_format_separate_lines_successfully(self):
+        f = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--sources', 'X-Repolib-Name: Google Chrome',
+                          'Enabled: yes',
+                          'Types: deb',
+                          'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
+                          'Suites: stable',
+                          'Components: main',
+             '--filename', 'intel-gpu-jammy.list'])
+        self.assertEqual(f.gpgKeyUri, None)
+        self.assertEqual(f.gpgKeyName, None)
+        self.assertEqual(f.sources, ['X-Repolib-Name: Google Chrome',
+                                     'Enabled: yes',
+                                     'Types: deb',
+                                     'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
+                                     'Suites: stable',
+                                     'Components: main'])
+        self.assertEqual(f.filename, 'intel-gpu-jammy.list')
+
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
+    def test_raise_application_add_with_only_one_gpgKeyUri_param(self, m_connect):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--gpgKeyUri', 'https://repositories.intel.com/gpu/intel-graphics.key',
+             '--sources', 'deb http://example.com/ focal main restricted universe',
+             'deb-src http://example.com/ focal-security main',
+             '--filename', 'intel-gpu-jammy.list'])
+        with pytest.raises(InbcException,
+                           match="Source requires either both gpgKeyUri and gpgKeyName "
+                                 "to be provided, or neither of them."):
+            Inbc(p, 'source', False)
+
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
+    def test_raise_application_add_with_only_one_gpgKeyName_param(self, m_connect):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--gpgKeyName', 'intel-graphics.gpg',
+             '--sources', 'deb http://example.com/ focal main restricted universe',
+             'deb-src http://example.com/ focal-security main',
+             '--filename', 'intel-gpu-jammy.list'])
+        with pytest.raises(InbcException,
+                           match="Source requires either both gpgKeyUri and gpgKeyName "
+                                 "to be provided, or neither of them."):
+            Inbc(p, 'source', False)
```

Comment:
**Suggestion:** Use a data-driven approach for tests to reduce code duplication and improve maintainability. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
    def test_raise_application_add_with_missing_gpg_params(self, m_connect):
        test_cases = [
            (['--gpgKeyUri', 'https://repositories.intel.com/gpu/intel-graphics.key'], 'gpgKeyName'),
            (['--gpgKeyName', 'intel-graphics.gpg'], 'gpgKeyUri')
        ]
        for args, missing_param in test_cases:
            with self.subTest(missing_param=missing_param):
                p = self.arg_parser.parse_args(['source', 'application', 'add'] + args + [
                    '--sources', 'deb http://example.com/ focal main restricted universe',
                    'deb-src http://example.com/ focal-security main',
                    '--filename', 'intel-gpu-jammy.list'])
                with pytest.raises(InbcException,
                                   match=f"Source requires either both gpgKeyUri and {missing_param} "
                                         "to be provided, or neither of them."):
                    Inbc(p, 'source', False)
```